packages/runtime/datastore/src/localChannelContext.ts

```diff
@@ -72,11 +72,12 @@ export class LocalChannelContext implements IChannelContext {
                 blobMap,
             );
         });
-        this.factory = registry.get(type);
-        if (this.factory === undefined) {
-            throw new Error(`Channel Factory ${type} not registered`);
-        }
         if (snapshotTree === undefined) {
```

Comment:
It's a bit weird to have a requirement that either snapshot or type is provided. Maybe we can have a base class (without such requirements) and two derived classes - one that takes type in constructor, and another - that takes snapshot? Such that it's obvious from types that you have to have one of them (as right now signature of constructor suggest that both are optional).